src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs

```diff
@@ -970,29 +925,20 @@ protected override async ValueTask DisposeAsyncCore()
         }
 
         /// <summary>
-        /// Creates a new JSON writer of <see cref="IJsonWriter"/> that can write
-        /// directly to the output stream. Returns null if unable to create
-        /// such a writer with the given constraints.
+        /// Initializes the output stream.
         /// </summary>
-        /// <param name="factory">The factory used to create the <see cref="IJsonWriter"/> instance.</param>
-        /// <param name="outputStream">The output stream to write to.</param>
-        /// <param name="isIeee754Compatible">True if the writer should write decimals and longs as strings.</param>
-        /// <param name="encoding">The text encoding of the output data.</param>
-        /// <returns>The JSON writer instance, or null if none could be created.</returns>
-        private static IJsonWriter CreateJsonWriter(
-            IStreamBasedJsonWriterFactory factory,
-            Stream outputStream,
-            bool isIeee754Compatible,
-            Encoding encoding)
+        /// <param name="stream">The supplied message stream.</param>
+        /// <returns>The output stream.</returns>
+        private Stream InitializeOutputStream()
         {
-            IJsonWriter jsonWriter = factory.CreateJsonWriter(outputStream, isIeee754Compatible, encoding);
-
-            if (jsonWriter == null)
+            if (this.Synchronous)

```

Comment:
I think the fact that we do something different based on whether we're in sync or async mode would further complicate my previous suggestion of encapsulating this buffer stream in the `DefaultJsonWriterFactory`. Feel free to #wontfix the suggestion if it distracts from the scope.